### DISCUSSION TITLE
How is Telemetry excluded when built manually

### DISCUSSION THREAD

**Discussion by BraINstinct0** [microsoft/winget-cli#1743] [Q&A] [answered]
Any code reference to the 'switch',
or what's different about the Azure build pipeline would be welcome.

**Comment by JohnMcPMS** [comment, marked as answer]
Our internal build does something (I'm not actually sure exactly what) to mark our events as worthy of collection.  Without that, they are just regular ETL tracelogging events that you can see on your device, but will never leave it.

**Reply by florelis** [reply]
The telemetry events are configured by [MicrosoftTelemetry.h](https://github.com/microsoft/winget-cli/blob/0a6cb30ecfa15f9fe81b50496f5b1f13f0644af1/src/AppInstallerCommonCore/Telemetry/MicrosoftTelemetry.h). From that file:

     Different versions of this file use different definitions for the
     TraceLoggingOption configuration macros. The definitions in this file are
     empty. As a result, providers using this configuration file will not join
     any ETW Provider Groups and will not be given any special treatment by
     group-sensitive ETW listeners.

During official builds, this file is replaced by one with special `TraceLoggingOption` macros that mark the events as worthy of collection (subject to the machine's settings).
